test/functional/feature_discover.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test -discover command."""
+
+import socket
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+
+def is_valid_ipv4_address(address):
+    try:
+        socket.inet_aton(address)
+    except socket.error:
+        return False
+    return True
+
+
+def is_valid_ipv6_address(address):
+    try:
+        socket.inet_pton(socket.AF_INET6, address)
+    except socket.error:
+        return False
+    return True
+
+
+class DiscoverTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
```

Comment:
On my system (MacOS 10.15.7) `localaddresses` is always empty. Which is a different behavior from what I see at `test/functional/feature_bind_port_discover.py` where `localaddresses` is accurate.

What I think is the problem is that without explicitly set `bind_to_localhost_only = False` [the node binds to localhost](https://github.com/brunoerg/bitcoin/blob/2022-02-discover-test/test/functional/test_framework/test_framework.py#L470-L471) which [is NET_UNROUTABLE](https://github.com/brunoerg/bitcoin/blob/2022-02-discover-test/src/test/netbase_tests.cpp#L47) thus [never added to mapLocalHost](https://github.com/brunoerg/bitcoin/blob/2022-02-discover-test/src/net.cpp#L286-L313) which is what is [returned for `localaddresses` by the `getnetworkinfo` RPC call](https://github.com/brunoerg/bitcoin/blob/2022-02-discover-test/src/rpc/net.cpp#L649-L661). With this change, `localaddresses` always contains my addresses as expected. 

```suggestion
    def set_test_params(self):
        self.setup_clean_chain = True
        self.bind_to_localhost_only = False
        self.num_nodes = 1
```

But I still don't understand how this works for others, so there must be a flaw in my thinking.

Reply:
Make sense, `bind_to_localhost_only`is True by default. thanks!